lib/membrane/rtp/packet.ex

```diff
@@ -1,89 +1,93 @@
 defmodule Membrane.RTP.Packet do
   @moduledoc """
-  Defines a struct describing an RTP packet and a way to parse it.
+  Defines a struct describing an RTP packet and a way to parse and serialize it.
   Based on [RFC3550](https://tools.ietf.org/html/rfc3550#page-13)
   """
 
-  alias Membrane.RTP.{Header, Packet}
+  alias Membrane.RTP.{Header, Utils}
 
   @type t :: %__MODULE__{
           header: Header.t(),
           payload: binary()
         }
 
-  defstruct [
-    :header,
-    :payload
-  ]
+  @enforce_keys [:header, :payload]
+  defstruct @enforce_keys
+
+  @spec serialize(t, align_to: pos_integer()) :: binary
+  def serialize(%__MODULE__{} = packet, [align_to: align_to] \\ [align_to: 1]) do
+    %__MODULE__{header: header, payload: payload} = packet
+    %Header{version: 2} = header
+    has_padding = 0
+    has_extension = if header.extension, do: 1, else: 0
+    marker = if header.marker, do: 1, else: 0
+    csrcs = Enum.map_join(header.csrcs, &<<&1::32>>)
+
+    serialized =
+      <<header.version::2, has_padding::1, has_extension::1, length(header.csrcs)::4, marker::1,
+        header.payload_type::7, header.sequence_number::16, header.timestamp::32, header.ssrc::32,
+        csrcs::binary, serialize_header_extension(header.extension)::binary, payload::binary>>
+
+    case Utils.align(serialized, align_to) do
+      {serialized, 0} ->
+        serialized
+
+      {serialized, _padding} ->
+        <<pre::2, _has_padding::1, post::bitstring>> = serialized
+        <<pre::2, 1::1, post::bitstring>>
+    end
+  end
 
-  @type error_reason() :: :wrong_version | :packet_malformed
+  defp serialize_header_extension(nil) do
+    <<>>
+  end
 
-  @spec parse(binary()) :: {:ok, Packet.t()} | {:error, error_reason()}
-  def parse(<<version::2, _::6, _::binary>>) when version != 2,
-    do: {:error, :wrong_version}
+  defp serialize_header_extension(%Header.Extension{data: data} = extension)
+       when byte_size(data) |> rem(4) == 0 do
```

Comment:
I would add one more function clause for case when data size is not divisable by 4 and warn inside it. In other case we will get an error here

Reply:
If it's not divisible, we cannot serialize the header in a way it can be parsed, so that's an error. We can possibly return `:ok | :error` tuples

Reply:
Okay, so let's merge it :+1: